### ISSUE TITLE
Download Fabric SDK for iOS when plugin is added.

### ISSUE THREAD

**Issue by Justin-Credible** [sarriaroman/FabricPlugin#22]
As suggested in https://github.com/sarriaroman/FabricPlugin/issues/4#issuecomment-196439447 we should download the Fabric SDK for iOS from Fabric's servers when the plugin is added.

This has several advantages:
* Consistent with Android (which obtains the SDK via Gradle)
* Avoids potential licensing issues (re: distribution of Fabric's binaries)
* Makes it easier to update the plugin (update download URL and hashes)

Changes Required:
- [ ] Add `downloadFile(...)` to `hooks/lib/utilities.js`
- [ ] Add `before_plugin_install` hook to download files from
  - https://kit-downloads.fabric.io/cocoapods/fabric/1.6.5/fabric.zip
  - https://kit-downloads.fabric.io/cocoapods/crashlytics/3.7.0/crashlytics.zip
- [ ] After download, calculate checksums on files to ensure they are what we're expecting
- [ ] Unzip files to a temporary location, verify the iOS framework bundles exist and then copy to:
  - `lib/ios/Crashlytics.framework`
  - `lib/ios/Fabric.framework`
- [ ] Ensure the existing `plugin.xml` references pick up the extracted files

**Comment by Justin-Credible** [comment]
@sarriaroman You can assign this one to me.

**Comment by sarriaroman** [comment]
I'm adding you as collaborator of the project ;)

**Comment by nbdamian** [comment]
Is there no way we can download the files latest version?
The way proposed here, will require an update to the plugin whenever a new fabric/crashlytics version is released.

**Comment by sarriaroman** [comment]
Yes and I think that is the right way to proceed to avoid problems with new APIs coming from the plugin, we will need to prevent automatic upgrades that bring problems into an application.

**Comment by emcniece** [comment]
I am not convinced that this is a good idea. 

Controlling package versions is important for stability. Explicit version updating gives developers the means to troubleshoot problems between versions, and Fabric/Crashlytics versions are effectively obfuscated by `cordova-fabric-plugin`. If either package updates during a platform reinstall or update, it may not be clear to the developer what exactly has changed - and this might lead to unnecessary debug.

If this is truly a danger, then `cordova-fabric-plugin` could "lock" packages to a certain version. The packages could still be obtained during platform add/update as needed, but a potential issue here is that some developers use a package cache for trusted sources. NPM packages might be cached inside of a controlled network, and if this plugin requires external connections to download Fabric/Crashlytics then developers may not be able to use this plugin within a restricted environment.

As it stands right now, including Fabric and Crashlytics right here in the repo avoids both of these problems. The downside here is that the repo must be maintained.

On a side note, I have added a script to make updating Fabric and Crashlytics a bit easier: https://github.com/sarriaroman/FabricPlugin/pull/112
